src/components/screens/DraggableItem.tsx

```diff
@@ -0,0 +1,212 @@
+      isGestureActive.value = true;
+      translateX.value = ctx.x + translationX;
+      translateY.value = ctx.y + translationY;
+
+      // 1. We calculate where the tile should be
+      const newOrder = getOrder(
+        translateX.value,
+        translateY.value,
+        Object.keys(positions.value).length - 1
+      );
+
+      // 2. We swap the positions
+      const oldOlder = positions.value[id];
+
+      if (newOrder !== oldOlder) {
+        const idToSwap = Object.keys(positions.value).find(
+          (key) => positions.value[key] === newOrder
+        );
+
+        if (idToSwap) {
+          // Spread operator is not supported in worklets
+          // And Object.assign doesn't seem to be working on alpha.6
+          const newPositions = JSON.parse(JSON.stringify(positions.value));
+          newPositions[id] = newOrder;
+          newPositions[idToSwap] = oldOlder;
+          positions.value = newPositions;
+        }
+      }
+
+      // 3. Scroll up and down if necessary
+      const lowerBound = scrollY.value;
+      const upperBound = lowerBound + containerHeight - tileSize;
+      const maxScroll = contentHeight - containerHeight;
+      const leftToScrollDown = maxScroll - scrollY.value;
+
+      if (translateY.value < lowerBound) {
+        const diff = Math.min(lowerBound - translateY.value, lowerBound);
+        scrollY.value -= diff;
+        scrollTo(scrollView, 0, scrollY.value, false);
+        ctx.y -= diff;
+        translateY.value = ctx.y + translationY;
+      }
+
+      if (translateY.value > upperBound) {
+        const diff = Math.min(translateY.value - upperBound, leftToScrollDown);
+        scrollY.value += diff;
+        scrollTo(scrollView, 0, scrollY.value, false);
+        ctx.y += diff;
+        translateY.value = ctx.y + translationY;
+      }
+    },
+    onEnd: () => {
+      const newPosition = getPosition(positions.value[id]);
+
+      translateX.value = withTiming(newPosition.x, animationConfig, () => {
+        isGestureActive.value = false;
+        runOnJS(onDragEnd)(positions.value);
+      });
+      translateY.value = withTiming(newPosition.y, animationConfig);
+    }
+  });
+
+  const animatedStyle = useAnimatedStyle(() => {
+    const zIndex = isGestureActive.value ? 100 : 0;
+    const scale = withSpring(isGestureActive.value ? 1.05 : 1);
+
+    return {
+      height: tileSize,
+      transform: [{ translateX: translateX.value }, { translateY: translateY.value }, { scale }],
+      width: tileSize,
+      zIndex
+    };
+  });
+
+  return (
+    <Animated.View style={[StyleSheet.absoluteFillObject, animatedStyle]}>
+      <PanGestureHandler enabled onGestureEvent={onGestureEvent}>
+        <Animated.View
+          style={[
+            StyleSheet.absoluteFillObject,
+            styles.draggableItem,
+            !!children?.props?.item?.tile && styles.noBorder
+          ]}
+        >
+          {children}
+        </Animated.View>
```

Comment:
the items are not exactly centred.

![Simulator Screen Shot - iPad Pro (11-inch) (4th generation) - 2022-11-21 at 10 20 28](https://user-images.githubusercontent.com/11755668/203018744-ef71458b-94d2-4101-be33-e7807b301ade.png)

Reply:
This device is an iPad 11".